sound/soc/sof/intel/hda-codec.c

```diff
@@ -124,10 +123,13 @@ static int hda_codec_probe(struct snd_sof_dev *sdev, int address,
 	if (hda_codec_use_common_hdmi ||
 	    (resp & 0xFFFF0000) != IDISP_VID_INTEL) {
 		hdev->type = HDA_DEV_LEGACY;
-		hda_codec_load_module(&hda_priv->codec);
+		ret = hda_codec_load_module(&hda_priv->codec);
+		 /* -1 = no device, 0 = no driver -> both errors */
+		if (!ret)
```

Comment:
that should be if (ret <= 0) if I understand the comment? the case with -1 would not be an error

Reply:
@plbossart I want to pass negative error codes as is from hda_codec_load_module() here. E.g. it can return -ENODEV or -ENOMEM. Only in the case of zero as return value (no driver attached), we return -ENOENT. This way you can see from the dbg traces which error case this is.

Reply:
ok, but then the comment can be improved. what looks like a bug is an intended feature...

ret = 0 means no driver, so explicitly adjust the return value. Negative values mean no device and will be returned as is.

Reply:
@plbossart ack, you are right, I'm trying to be too clever here and readability suffers. I'll update and resubmit

Reply:
@plbossart Now updated. This not 100% happy, but at least it's better. I realized we pass also positive values unmodified, so adjusted the language a bit more. Let's iterate until it's understandable.